src/libYARP_dev/src/devices/JoypadControlServer/JoypadControlServer.cpp

```diff
@@ -0,0 +1,742 @@
+            }
+        }
+        if(write)m_portAxis.write();
+    }
+
+    if (m_portTrackball.valid)
+    {
+        unsigned int count;
+        bool         write;
+
+        write = true;
+        if(m_device->getTrackballCount(count))
+        {
+            Bottle& b = m_portTrackball.prepare();
+            b.clear();
+            for(size_t i = 0; i < count; ++i)
+            {
+                yarp::sig::Vector v;
+                if(!m_device->getTrackball(i, v))
+                {
+                    yError() << "cannot get axis with id" << i;
+                    write = false;
+                    break;
+                }
+                Bottle& ball = b.addList();
+                for(size_t j = 0; j < v.size(); ++i)
+                {
+                    ball.addDouble(v[j]);
+                }
+            }
+        }
+        if(write)m_portTrackball.write();
+    }
+
+    if (m_portStick.valid)
+    {
+        unsigned int count;
+        bool         write;
+
+        write = true;
+        if(m_device->getStickCount(count))
+        {
+            Bottle& b = m_portStick.prepare();
+            b.clear();
+            for(size_t i = 0; i < count; ++i)
+            {
+                Vector       v;
+                unsigned int dofCount;
+                if(!m_device->getStick(i, v, m_coordsMode) || !m_device->getStickDoF(i, dofCount) || v.size() != dofCount)
+                {
+                    write = false;
+                    break;
+                }
+                Bottle& stick = b.addList();
+                for(size_t j = 0; j < dofCount; ++i)
+                {
+                    stick.addDouble(v[j]);
+                }
+            }
+        }
+        if(write)m_portStick.write();
+    }
+
+    if (m_portStick.valid)
+    {
+        unsigned int count;
+        bool         write;
+
+        write = true;
+        if(m_device->getTouchSurfaceCount(count))
+        {
+            Bottle& b = m_portTouch.prepare();
+            b.clear();
+            for(unsigned int i = 0; i < count; ++i)
+            {
+                Vector       v;
+                if(!m_device->getTouch(i, v))
+                {
+                    write = false;
+                    break;
+                }
+                Bottle& touch = b.addList();
+                for(size_t j = 0; j < v.size(); ++i)
+                {
+                    touch.addDouble(v[j]);
+                }
+            }
+        }
+        if(write)m_portTouch.write();
+    }
+}
+
+bool JoypadControlServer::attachAll(const PolyDriverList& p)
+{
+    if (p.size() != 1)
+    {
+        yError("JoypadControlServer: cannot attach more than one device");
+        return false;
+    }
+
+    yarp::dev::PolyDriver* Idevice2attach = p[0]->poly;
+    if(p[0]->key == "IJoypadController")
+    {
+        yInfo() << "JoypadControlServer: Good name Dude!";
+    }
+    else
+    {
+        yInfo() << "JoypadControlServer: Bad name Dude!!";
+    }
+
```

Comment:
:joy:
Open question: will it be possible in the futurefor the wrapper to attaches to multiple joystick device drivers?